### PR TITLE
update attributesToCrop and attributesToHighlight

### PR DESCRIPTION
**Pull request by qdequele**
Improved documentation on:

- AttributesToHighlight "*" behaviour
- AttributesToCrop: accepts as parameter: `<attributes>`, `<attributes:croplength>`, `"*"`
- CropLength: More precise documentation
- AttributesToRetrieve: improved documentation on default value

### CODE REVIEW

File: guides/advanced_guides/search_parameters.md [hunk 1]
```diff
@@ -149,7 +162,9 @@ You will get the following response with the **cropped version in the \_formatte
 
 `cropLength=<Integer>`
```
Comment (matboivin) [inline guides/advanced_guides/search_parameters.md:163]:
```suggestion
`cropLength=<Integer>` (Optional, positive integer, defaults to `200`)
```

File: guides/advanced_guides/search_parameters.md [hunk 2]
```diff
@@ -149,7 +162,9 @@ You will get the following response with the **cropped version in the \_formatte
...
 
-Number of characters to keep on each side of the start of the matching word. See [attributesToCrop](/guides/advanced_guides/search_parameters.md#attributes-to-crop)
+The default value is 200.
```
Comment (matboivin) [inline guides/advanced_guides/search_parameters.md:165]:
```suggestion
```

File: guides/advanced_guides/search_parameters.md [hunk 3]
```diff
@@ -101,13 +103,24 @@ $ curl -X GET -G 'http://localhost:7700/indexes/movies/search' \
...
 Attributes whose values will be cropped if they contain a matched query word.
 
-`attributesToCrop=<Attribute>,<Attribute>,...`
+`attributesToCrop=<Attribute:Croplength>,<Attribute:Croplength>,...`
 
-- `<Attribute>` (Optional, string, defaults to empty)
+- `<Attribute:Croplength>` OR `<Attribute>` (Optional, string, defaults to empty)
+
+  Comma-separated list of attributes whose values will be cropped if they contain a matched query word.<br>
+  Each attribute can be joined by an optional croplength.
```
Comment (matboivin) [inline guides/advanced_guides/search_parameters.md:111]:
```suggestion
  Each attribute can be joined by an optional `cropLength`.
```

File: guides/advanced_guides/search_parameters.md [hunk 4]
```diff
@@ -101,13 +103,24 @@ $ curl -X GET -G 'http://localhost:7700/indexes/movies/search' \
...
+
+- `"*"`
+
+  The attribute can also be `*`. In that case, `*` will be replaced by all attributes present in `attributesToRetrieve`.
```
Comment (matboivin) [inline guides/advanced_guides/search_parameters.md:115]:
```suggestion
  The `*` character can also be used. In that case, all the attributes present in `attributesToRetrieve` will be assigned to `attributesToCrop`.
```

File: guides/advanced_guides/search_parameters.md [hunk 5]
```diff
@@ -101,13 +103,24 @@ $ curl -X GET -G 'http://localhost:7700/indexes/movies/search' \
 
 Attributes whose values will be cropped if they contain a matched query word.
 
-`attributesToCrop=<Attribute>,<Attribute>,...`
+`attributesToCrop=<Attribute:Croplength>,<Attribute:Croplength>,...`
```
Comment (matboivin) [inline guides/advanced_guides/search_parameters.md:106]:
```suggestion
You have two options:

`attributesToCrop=<Attribute:Croplength>,<Attribute:Croplength>,...`
```

File: guides/advanced_guides/search_parameters.md [hunk 6]
```diff
@@ -101,13 +103,24 @@ $ curl -X GET -G 'http://localhost:7700/indexes/movies/search' \
...
 
-- `<Attribute>` (Optional, string, defaults to empty)
+- `<Attribute:Croplength>` OR `<Attribute>` (Optional, string, defaults to empty)
+
+  Comma-separated list of attributes whose values will be cropped if they contain a matched query word.<br>
+  Each attribute can be joined by an optional croplength.
+
```
Comment (matboivin) [inline guides/advanced_guides/search_parameters.md:112]:
```suggestion

`attributesToCrop=*`

```

File: guides/advanced_guides/search_parameters.md [hunk 7]
```diff
@@ -101,13 +103,24 @@ $ curl -X GET -G 'http://localhost:7700/indexes/movies/search' \
...
+- `"*"`
```
Comment (matboivin) [inline guides/advanced_guides/search_parameters.md:113]:
```suggestion
- `*`
```

File: guides/advanced_guides/search_parameters.md [hunk 8]
```diff
@@ -83,6 +83,8 @@ Attributes to **display** in the returned documents.
 
 `attributesToRetrieve=<Attribute>,<Attribute>,...`
 
+The default value is all the attributes in the [displayedAttributes](/guides/advanced_guides/settings.md#displayed-attributes) list present in the [settings](/guides/advanced_guides/settings.md) (which by default contains all the attributes found in every document).
+
 - `<Attribute>` (Optional, string, Defaults to `*`)
 
   Comma-separated list of attributes whose fields will be present in the returned documents.
```
Comment (matboivin) [inline guides/advanced_guides/search_parameters.md:90]:
```suggestion
  Comma-separated list of attributes whose fields will be present in the returned documents.
  
  Defaults to the attributes added to the [displayedAttributes list](/guides/advanced_guides/settings.md#displayed-attributes) which is a [configuration setting](/guides/advanced_guides/settings.md#displayed-attributes) that contains by default all attributes found in the documents.
```

File: guides/advanced_guides/search_parameters.md [hunk 9]
```diff
@@ -83,6 +83,8 @@ Attributes to **display** in the returned documents.
 
 `attributesToRetrieve=<Attribute>,<Attribute>,...`
 
+The default value is all the attributes in the [displayedAttributes](/guides/advanced_guides/settings.md#displayed-attributes) list present in the [settings](/guides/advanced_guides/settings.md) (which by default contains all the attributes found in every document).
```
Comment (matboivin) [inline guides/advanced_guides/search_parameters.md:86]:
```suggestion
```

File: guides/advanced_guides/search_parameters.md [hunk 10]
```diff
@@ -161,6 +175,17 @@ Attributes whose values will contain **highlighted matching query words**.
 
   Comma-separated list of attributes. Every matching query words in the given attribute field will be wrapped around an `<em>` tag.
 
+- `"*"`
+
+  The attribute can also be `*`. In that case, `*` will be replaced by all attributes present in `attributesToRetrieve`.
```
Comment (bidoubiwa) [inline guides/advanced_guides/search_parameters.md:180]: Do you want to change it here aswel @matboivin ?

Comment (bidoubiwa) [reply guides/advanced_guides/search_parameters.md:180]:
```suggestion
 The `*` character can also be used. In that case, all the attributes present in `attributesToRetrieve` will be assigned to `attributesToHighlight`.
```

**Pull request merged**